stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4797,7 +4797,7 @@ ParseResult parseConvolutionDimensions(AsmParser& parser,
     }
 
     llvm::SmallDenseMap<int64_t, int64_t> spatialDimsMap;
-    constexpr int64_t kInvalidDimension = -1;
```

Comment:
It looks like in all the instances covered in this PR, `-1` is used as a legit sentinel value for a specific algorithm (axis number here and input number below). I think that it makes all of them out of scope for the `kDynamicSize` change.

The motivation behind #327 is to fix all the situations where `-1` is used as a dimension size to indicate that the size is dynamic. In those situations, `kDynamicSize` should be used instead to make code robust (because `kDynamicSize` is going to change to a different value soon, and all the uses of `-1` in that capacity will become incorrect).

In other situations (like the ones in this PR), I think we should continue using `-1`. E.g. I would be surprised if something called `kInvalidDimension` was equal to `kDynamicSize`, because the former is a dimension number and the latter is a dimension size.

What do you think?

Reply:
Understood now. The other notable places that use `-1` come from td files (some comments use them, which we will most likely overwrite when we get to those ops, at least in StablehloOps.td, and a few other similar occurrences in ChloOps.td) and mlir files (where the `?` dimension size is hard coded as -1).

Then based on my understanding, I should be changing the mlir files containing the -1 dimension size, but I'm not sure how I can turn this into kDynamicSize. I might need a bit more context to resolving this issue. Or should I be looking somewhere else entirely?